### ISSUE TITLE
Hover over system settings and reboot?

### ISSUE THREAD

**Issue by irowiki** [zoogie/MSET9#22]
Hey Zoogie, we were wondering what the purpose of selecting system settings and rebooting, then entering system settings was, as it seems to work when entering system settings from any other app or position on the home screen?

**Comment by zoogie** [comment]
Warning: fairly technical response.

When you first boot the 3ds, you start the system (and its various RAM regions) in a known/predictable state. The more you "do stuff" like move the home menu cursor around, the more the RAM regions will change as the CPU processes data and interacts with RAM (aka entropy).

Since the exploit relies on human behavior (the specific user menu navigation through System Settings) to place a certain jump address in r5 before the crucial blx r5 to ID1, the "do stuff" has to be kept to a minimum to ensure reliability. And yes, I've found out by experimentation that moving the cursor around prior to launching System Settings will corrupt the exploit process (increased entropy).

tl;dr - the purpose of coldboot to Sys Settings is to ensure stability in what is a delicate internal codepath to a successful exploit.

-----------
Note: Does there exist an easier "do stuff" that could also lead to a successful exploit state? Possibly! And I hope someone finds one, but after a year of experiments after finding the underlying bug, I haven't found one.

**Comment by zoogie** [comment]
Addendum:
"to place a certain jump address in r5 before the crucial blx r5 to ID1" -  The arm9 memory area in which this "certain jump address" lies is uninitialized (this is the actual bug). Since uninitialized data contains "whatever" data just happens to be there (there are no rules) we can use user input/behavior to try to place a good jump address in that spot. Obviously, we don't have full control of what goes there, but we can get lucky sometimes. And we did.

**Comment by irowiki** [comment]
Thank you for the great explanation, and it actually makes sense to me.

I think a wording change will help as people get confused by "hover", like "use the dpad to select system settings" 

I'll go back to the discord and chat them up about it.
